rustls/src/anchors.rs

```diff
@@ -95,6 +95,18 @@ impl RootCertStore {
         self.roots.len()
     }
 
+    fn add_internal(&mut self, der: &[u8]) -> Result<(), Error> {
+        let ta = webpki::TrustAnchor::try_from_cert_der(der)
+            .map_err(|_| Error::InvalidCertificate(CertificateError::BadEncoding))?;
+        let ota = OwnedTrustAnchor::from_subject_spki_name_constraints(
+            ta.subject,
+            ta.spki,
+            ta.name_constraints,
+        );
+        self.roots.push(ota);
+        Ok(())
+    }
```

Comment:
I extracted the code of `RootCertStore::add()` here to not have to duplicate it in `RootCertStore::add_parsable_certificates()`. I couldn't use `RootCertStore::add()` directly because it requires owned bytes.

Let me know if you want me to handle it differently.

Reply:
I think extracting this makes sense. I'd like to have this added below the other relevant methods rather than before them. Maybe inline the `ota` binding?

Reply:
Done.